File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Set<Character> set=new HashSet<>();
        int length=0;
        int e=0;
        for(int f=0;f<s.length();f++){
            if(set.contains(s.charAt(f))){
                while(!(s.charAt(f)==s.charAt(e))){
                    set.remove(s.charAt(e));
                    e++;
                }
                set.remove(s.charAt(e));
                e++;
                set.add(s.charAt(f));
            }else{
                set.add(s.charAt(f));
                length=Math.max(length,f-e+1);
            }
        }
        return length;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n=s.size();
        unordered_set<char> mySet;
        int result=0;
        int left=0;
        for(int e=0;e<n;e++){
            if(mySet.find(s[e])==mySet.end()){
                mySet.insert(s[e]);
                result=max(result,(e-left)+1);
            }else{
                while(mySet.find(s[e])!=mySet.end()){
                    mySet.erase(s[left]);
                    left++;
                }
                mySet.insert(s[e]);
                result=max(result,(e-left)+1);
            }
        }
        return result;
    }
};